src/featbin/wav-reverberate.cc

```diff
@@ -0,0 +1,250 @@
+// featbin/extract-segments.cc
```

Comment:
Please add a small unit test to confirm that the block convolution program generates the same results as the full signal convolution.